src/shared/components/MapBlock.js

```diff
@@ -71,6 +97,20 @@ MapBlock.defaultProps = {
   ]
 };
 
+const mapStateToProps = (state) => {
+  return {
+    cards: state.map.cards
+  };
+};
+
+const mapDispatchToProps = (dispatch) => {
+  return {
+    mapClick: () => {
+      dispatch(addCard());
+    }
+  };
+};
+
 MapBlock = controllable(MapBlock, ['center', 'zoom', 'markers']);
 
-export default MapBlock;
+export default connect(mapStateToProps, mapDispatchToProps)(MapBlock);
```

Comment:
connect is a role for container. Please extract out line 100 - 116 as a container.